thali/thalireplicationmanager.js

```diff
@@ -38,25 +47,54 @@ ThaliReplicationManager.events = {
 };
 
 /**
-* Starts the Thali replication manager with the given device name and port number
-* @param {String} deviceName the device name to advertise.
+* Returns the current device's id if available. If not, the provided callback
+* function is saved and called when the id does become available.
+* @param {Function} cb the callback which returns the current device's id.
+*/
+ThaliReplicationManager.prototype.getDeviceIdentity = function (cb) {
+  if(this._deviceIdentityFlag == deviceIdentityFlag.deviceIdentityAvailable) {
+    cb(null, this._deviceName);
```

Comment:
It is generally considered best coding practice to avoid implementing the equivalent of case statements as nested if-then-elses. It is hard to read and keep track of state. A much cleaner solution, in my opinion, is to just use return statements. In other words

if (..deviceIdentityAvailable...) {
  cb();
  return;
}

if (...noDeviceIdentitySet...) {
  stuff...
  return;
}

if (...gettingDeviceIdentity...) {
  blah...
  return;
}